WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/CreateSiteUseCase.kt

```diff
@@ -45,12 +49,18 @@ class CreateSiteUseCase @Inject constructor(
          * time of this comment, changing FluxC's Payload might end up affecting the old site creation flow,
          * so the workaround is applied here instead.
          */
-        val domain = if (isWordPressComSubDomain(siteData.domain)) {
-            urlUtilsWrapper.extractSubDomain(siteData.domain)
-        } else siteData.domain
+        val domain = when {
+            siteData.domain.isNullOrEmpty() -> null
+            isWordPressComSubDomain(siteData.domain) -> urlUtilsWrapper.extractSubDomain(siteData.domain)
+            else -> siteData.domain
+        }
+        val username = accountStore.account?.userName
+        val emailUser = signupUtils.createUsernameFromEmail(accountStore.account.email)
```

Comment:
I'm curious when we'd have an email, but not have the username :thinking: . Is this mainly relevant for site creation during signup? If so, does creating the domain based on email have any privacy implications - enough to warrant a dialog / disclosure of some sort? Otoh, I understand we are trying to simplify the experience, so introducing additional messaging is a trade-off to consider :sweat_smile:

Reply:
Great catch @mkevins 👍 Thank you for bringing this up 🙇 
I got into using the email [while checking this other usage of the getuserName()](https://github.com/wordpress-mobile/WordPress-Android/blob/082b8f039aef4abf647e9b927bd3d3834137f25e/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java#L632) function. This is a completely different usecase though since it is the epilogue of the signup where the username is generated from the user provided email. My case is not the same since it is not used in a user facing form but the public url which is not a good idea since it creates a privacy issue 😬 
Investigating this further I came to the conclusion that the user will always have a username at this point and there is no need for the added logic.
I updated this with 31301744d733d18fee46f4a2a31fd435140f34e4

Reply:
Thanks for updating to address this :heart: